security-analysis/src/main/java/com/powsybl/security/Security.java

```diff
@@ -160,33 +163,49 @@ public static String printLimitsViolations(List<LimitViolation> violations) {
         List<LimitViolation> filteredViolations = filter.apply(violations);
-        if (filteredViolations.size() > 0) {
-            filteredViolations.sort(Comparator.comparing(LimitViolation::getSubjectId));
-            Table table = new Table(9, BorderStyle.CLASSIC_WIDE);
-            table.addCell("Country");
-            table.addCell("Base voltage");
-            table.addCell("Equipment (" + filteredViolations.size() + ")");
-            table.addCell("Violation type");
-            table.addCell("Violation name");
-            table.addCell("Value");
-            table.addCell("Limit");
-            table.addCell("abs(value-limit)");
-            table.addCell("Loading rate %");
-            for (LimitViolation violation : filteredViolations) {
-                table.addCell(violation.getCountry() != null ? violation.getCountry().name() : "");
-                table.addCell(Float.isNaN(violation.getBaseVoltage()) ? "" : Float.toString(violation.getBaseVoltage()));
-                table.addCell(violation.getSubjectId());
-                table.addCell(violation.getLimitType().name());
-                table.addCell(getViolationName(violation));
-                table.addCell(Float.toString(violation.getValue()));
-                table.addCell(getViolationLimit(violation));
-                table.addCell(Float.toString(Math.abs(violation.getValue() - violation.getLimit() * violation.getLimitReduction())));
-                table.addCell(Integer.toString(getViolationValue(violation)));
-            }
-            return table.render();
+
+        try (TableFormatter formatter = formatterFactory.create(writer,
+                "",
+                formatterConfig,
+                new Column("Country"),
+                new Column("Base voltage"),
+                new Column("Equipment (" + filteredViolations.size() + ")"),
+                new Column("Violation type"),
+                new Column("Violation name"),
+                new Column("Value"),
+                new Column("Limit"),
+                new Column("abs(value-limit)"),
+                new Column("Loading rate %"))) {
+            filteredViolations.stream()
+                    .sorted(Comparator.comparing(LimitViolation::getSubjectId))
+                    .forEach(violation -> {
+                        try {
+                            formatter.writeCell(violation.getCountry() != null ? violation.getCountry().name() : "")
+                                     .writeCell(Float.isNaN(violation.getBaseVoltage()) ? "" : Float.toString(violation.getBaseVoltage()))
+                                     .writeCell(violation.getSubjectId())
+                                     .writeCell(violation.getLimitType().name())
+                                     .writeCell(getViolationName(violation))
+                                     .writeCell(violation.getValue(), createNumberCellStyle(), Security.NUNBER_STRING_FORMAT)
```

Comment:
The CellStyle instance should be stored in a local variable to avoid multiple instance of the same object